src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  userTodo?: Todo | null;
+  setModalIcon: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const TodoModal: React.FC<Props> = ({ userTodo, setModalIcon }) => {
+  const [loadingModalIcon, setLoadingModalIcon] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  if (userTodo) {
+    getUser(userTodo.userId)
+      .then(setUser)
+      .finally(() => setLoadingModalIcon(false));
+  }
```

Comment:
Put this logic inside use effect to optimize component and don't fetch data on every rerender